src/main/java/mate/repository/impl/BookRepositoryImpl.java

```diff
@@ -0,0 +1,52 @@
+package mate.repository.impl;
+
+import java.util.List;
+import mate.model.Book;
+import mate.repository.BookRepository;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public class BookRepositoryImpl implements BookRepository {
+    private final SessionFactory sessionFactory;
+
+    @Autowired
+    public BookRepositoryImpl(SessionFactory sessionFactory) {
+        this.sessionFactory = sessionFactory;
+    }
```

Comment:
You have lombok library, lets use @requireadargsconstructor annotation instead of @Autowire